libd-llvm/src/d/llvm/backend.d

```diff
@@ -119,22 +117,13 @@ public:
 		auto targetData = LLVMGetTargetMachineData(targetMachine);
 		LLVMAddTargetData(targetData, pm);
 		LLVMPassManagerBuilderPopulateModulePassManager(pmb, pm);
-		LLVMRunPassManager(pm, dmodule);
-		
-		// Dump module for debug purpose.
-		// LLVMDumpModule(dmodule);
-		
-		/+
-		import std.stdio;
-		writeln("\nASM generated :");
-		
-		LLVMTargetMachineEmitToFile(targetMachine, dmodule, "/dev/stdout".ptr, LLVMCodeGenFileType.Assembly, &errorPtr);
-		// +/
-		
+		LLVMRunPassManager(pm, pass.dmodule);
+	}
+
+	void emitObject(string objFile) {
 		import std.string;
-		
 		char* errorPtr;
-		auto linkError = LLVMTargetMachineEmitToFile(targetMachine, dmodule, objFile.toStringz(), LLVMCodeGenFileType.Object, &errorPtr);
+		auto linkError = LLVMTargetMachineEmitToFile(targetMachine, pass.dmodule, objFile.toStringz(), LLVMCodeGenFileType.Object, &errorPtr);
```

Comment:
I meant, as opposed to use dmodule directly, but pass is not aliasthised here, so indeed.